### ISSUE TITLE
column reference is ambiguous on insert select with single table

### ISSUE THREAD

**Issue by danilobuerger** [sqlc-dev/sqlc#1328]
### Version

1.11.0

### What happened?

When using INSERT ... SELECT with a single table, sqlc is not able to compile.

### Relevant log output

```shell
column reference "name" is ambiguous
```

### Database schema

```sql
CREATE TABLE authors (
  id   BIGSERIAL PRIMARY KEY,
  name text      NOT NULL,
  bio  text
);
```

### SQL queries

```sql
INSERT INTO authors (id, name, bio) 
SELECT $1, name, bio
FROM authors
WHERE name = $2;
```

### Configuration

```yaml
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/3b568d39761e26955617d4db2b555fd6017b0fdcd6245147a73b57d5d48566cb

### What operating system are you using?

macOS

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by bongster** [comment]
i'm not sure but i think your declared author from insert and select are duplicated so when i use alias from select it's work 
https://play.sqlc.dev/p/052203b68a8267990ea741fbfe3abcb4ae6cc8efde9f6b0bcf423c064b135735

**Comment by danilobuerger** [comment]
@bongster the query I posted works with postgres, so it should work here too without having to define an alias.

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/insert_select_param/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/ef6ea73896e3202f0e843d53d37ea8d5e50ca8c90cf14edb4c4e264aad56a1c9)
